contracts/L2Forwarder.sol

```diff
@@ -74,11 +74,14 @@ contract L2Forwarder is L2ForwarderPredictor {
     /// @dev Bridge tokens to an L3 that uses ETH for fees. Entire ETH and token balance is sent.
     function _bridgeToEthFeeL3(L2ForwarderParams memory params) internal {
         // get balance and approve gateway
+        // @review - routerOrInbox is user input, whoever initiated the teleport can set it to anything and get approval
+        //         - currently teleport must use L2Forwarder that is owned by alias(msg.sender), so not a problem now
         uint256 tokenBalance = _approveGatewayForBalance(params.routerOrInbox, params.l2Token);
 
         // send tokens through the bridge to intended recipient
         // (send all the ETH we have too, we could have more than msg.value b/c of fee refunds)
         // overestimate submission cost to ensure all ETH is sent through
+        // @review - this might create some issue when multiple teleporter are simutanously initiated and some are failed
```

Comment:
i think the case that's weird is if there are 2 teleports of the same token with different `to` or something and one of them fails to call the factory. if they have all the same forwarder parameters then i think it's mostly fine. The forwarder address used to depend on all the parameters instead of only params.owner, i think that would fix the issue for the most part? I changed it back when removing the relayer feature reintroducing this problem

Reply:
we can also just make it not forward all eth, if excess was sent can be retrieved via rescue